src/libsync/bulkpropagatordownloadjob.cpp

```diff
@@ -112,14 +109,34 @@ bool BulkPropagatorDownloadJob::scheduleSelfOrChild()
 
     _state = Running;
 
-    for (const auto &fileToDownload : _filesToDownload) {
-        qCDebug(lcBulkPropagatorDownloadJob) << "Scheduling bulk propagator job:" << this << "and starting download of item"
-                                             << "with file:" << fileToDownload->_file << "with size:" << fileToDownload->_size;
-        _filesDownloading.push_back(fileToDownload);
-        start(fileToDownload);
+    for (const auto &fileToDownload : std::as_const(_filesToDownload)) {
+        Q_ASSERT(fileToDownload->_type == ItemTypeVirtualFileDehydration || fileToDownload->_type == ItemTypeVirtualFile);
+
+        if (propagator()->localFileNameClash(fileToDownload->_file)) {
+            _parentDirJob->appendTask(fileToDownload);
+            finalizeOneFile(fileToDownload);
+            qCCritical(lcBulkPropagatorDownloadJob) << "File" << QDir::toNativeSeparators(fileToDownload->_file) << "can not be downloaded because it is non virtual!";
+            abortWithError(fileToDownload, SyncFileItem::NormalError, tr("File %1 cannot be downloaded because it is non virtual!").arg(QDir::toNativeSeparators(fileToDownload->_file)));
+            return false;
+        }
     }
 
-    _filesToDownload.clear();
+    const auto &vfs = propagator()->syncOptions()._vfs;
+    Q_ASSERT(vfs && vfs->mode() == Vfs::WindowsCfApi);
+
+    const auto r = vfs->createPlaceholders(_filesToDownload);
```

Comment:
what should happen if this fails and returns an error string?

Reply:
done